launch/launch/launch_description_source.py

```diff
@@ -26,8 +33,10 @@ class LaunchDescriptionSource:
     def __init__(
         self,
         launch_description: Optional[LaunchDescription] = None,
-        location: str = '<string>',
+        location: SomeSubstitutionsType = '<string>',
         method: str = 'unspecified mechanism from a script',
+        *,
+        get_method: Optional[callable] = None
```

Comment:
@ivanpauno why not use method overloading in derived classes instead?

Reply:
Sounds good